cc/transport/grpc_unary_transport.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CC_TRANSPORT_GRPC_UNARY_TRANSPORT_H_
+#define CC_TRANSPORT_GRPC_UNARY_TRANSPORT_H_
+
+#include "absl/log/log.h"
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "cc/transport/transport.h"
+#include "cc/transport/util.h"
+#include "grpcpp/client_context.h"
+#include "proto/crypto/crypto.pb.h"
+#include "proto/session/messages.pb.h"
+#include "proto/session/service_unary.pb.h"
+
+namespace oak::transport {
+
+// Transport class for communication with unary gRPC Oak service. Evidence
+// must be collected from the enclave and verified prior to issuing any data
+// requests. This template class can be used to communicate with any unary
+// stubby Oak service that use a gRPC interface consistent with
+// oak/proto/session/service_unary.proto.
+template <typename OakBackendStub>
+class GrpcUnaryTransport : public ::oak::transport::TransportWrapper {
```

Comment:
Will `OakBackendStub` be used for an internal channel implementation and not only gRPC ? 

Then this class probably could just be called `UnaryTransport` (since it won't only be used for gRPC). And if it is only for gRPC, then discard this comment.

Reply:
The corresponding streaming version could also updated to become a template and renamed to `StreamingTransport` (but probably in a separate PR)

Reply:
So you are correct in that `OakBackendStub` could be used for the internal channel implementation as well, however calls to the server differ between the internal version and gRPC. If you wanted a single unary class, in the future maybe we could have an option in the constructor for clients to specify whether they want to use gRPC or a different form of communication? This then determines how requests to the server are called. Do you feel that is an idea worth pursuing?